File: python/fastapi-sqlalchemy/schemas/pet.py
from sqlalchemy import Column, ForeignKey, Integer, String, Date
from sqlalchemy.orm import relationship
from db.db import Base
from schemas.type_pet import TypePet
from schemas.breed import Breed


class Pet(Base):
    __tablename__ = "mascotas"

    id_pet = Column(
        name="codigomascota",
        type_=Integer,
        autoincrement="auto",
        nullable=False,
        primary_key=True,
    )
    name = Column(name="nombre", type_=String, nullable=False)
    id_type_pet = Column(
        Integer, ForeignKey("tiposMascotas.codigotipomascota"), name="codigotipomascota"
    )
    id_breed = Column(Integer, ForeignKey("razas.codigoraza"), name="codigoraza")
    check_in_date = Column(name="fechaingreso", type_=Date)
    birthday = Column(name="fechanacimiento", type_=Date)
    description = Column(name="observaciones", type_=String)

    type_pet = relationship("TypePet", back_populates="pets")
    breed = relationship("Breed", back_populates="pets")
    adoptions = relationship("Adoption", back_populates="pets")

File: javascript/express-pet-shelter/models/entities/pet.js
import { DataTypes } from "sequelize";
import { sequelize } from "../../db/db.js";
import { TypePet } from "./typePet.js";
import { Breed } from "./breed.js";

export const Pet = sequelize.define(
  "mascotas",
  {
    petId: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true,
      field: "codigomascota",
    },
    name: {
      type: DataTypes.STRING,
      field: "nombre",
    },
    birthday: {
      type: DataTypes.DATEONLY,
      field: "fechanacimiento",
    },
    typePet: {
      type: DataTypes.INTEGER,
      field: "codigotipomascota",
    },
    typeBreed: {
      type: DataTypes.INTEGER,
      field: "codigoraza",
    },
    dateAdmission: {
      type: DataTypes.DATEONLY,
      field: "fechaingreso",
    },
    observations: {
      type: DataTypes.STRING,
      field: "observaciones",
    },
  },
  {
    timestamps: false,
  }
);

Pet.belongsTo(TypePet, {
  foreignKey: "typePet",
  targetKey: "idTypePet",
});

TypePet.hasMany(Pet, {
  foreignKey: "typePet",
  sourceKey: "idTypePet",
});

Pet.belongsTo(Breed, {
  foreignKey: "typeBreed",
  targetKey: "idBreed",
});

Breed.hasMany(Pet, {
  foreignKey: "typeBreed",
  sourceKey: "idBreed",
});
